Dockerfile

```diff
@@ -1,21 +1,35 @@
-FROM php:7.1-fpm
+FROM php:7.2-fpm
 
-# Based on https://github.com/docker-library/wordpress/blob/d122aa204948aecd651af435fe4ca058bb99545d/php7.1/fpm/Dockerfile
+# Based on https://github.com/docker-library/wordpress/blob/master/php7.2/fpm/Dockerfile
 
 # install the PHP extensions we need
 RUN set -ex; \
+	\
+	savedAptMark="$(apt-mark showmanual)"; \
 	\
 	apt-get update; \
-	apt-get install -y \
+	apt-get install -y --no-install-recommends \
 		libjpeg-dev \
 		libpng-dev \
 	; \
-	rm -rf /var/lib/apt/lists/*; \
-	rm -rf /var/cache/apt/; \
 	\
 	docker-php-ext-configure gd --with-png-dir=/usr --with-jpeg-dir=/usr; \
-	docker-php-ext-install gd mysqli opcache
-# TODO consider removing the *-dev deps and only keeping the necessary lib* packages
+	docker-php-ext-install gd mysqli opcache zip; \
+	\
+# reset apt-mark's "manual" list so that "purge --auto-remove" will remove all build dependencies
+	apt-mark auto '.*' > /dev/null; \
+	apt-mark manual $savedAptMark; \
+	ldd "$(php -r 'echo ini_get("extension_dir");')"/*.so \
```

Comment:
This seems like a hacky way to do this. Also this is not very readable without taking the time to look at it.

Reply:
I stole it from the Wordpress repo